python/sidecar/http.py

```diff
@@ -32,11 +33,18 @@
 filesystem_api = FastAPI()  # API for interacting with the filesystem
 project_api = FastAPI()  # API for interacting with projects
 
+def update_envs_from_headers(request: Request) -> None:
+    os.environ["PROJECT_DIR"] = str(request.headers.get("X-PROJECT_DIR"))
+    os.environ["OPENAI_API_KEY"] = str(request.headers.get("X-OPENAI_API_KEY"))
+    os.environ["OPENAI_CHAT_MODEL"] = str(request.headers.get("X-OPENAI_CHAT_MODEL"))
+    os.environ["SIDECAR_ENABLE_LOGGING"] = str(request.headers.get("X-SIDECAR_ENABLE_LOGGING"))
+    os.environ["SIDECAR_LOG_DIR"] = str(request.headers.get("X-SIDECAR_LOG_DIR"))
```

Comment:
@gjreda I'm sending the env variables via "x-..." headers and implemented this utility method to update the os env vars when calling via HTTP so that the rest of the code (sidecar utilities) work as expected.

This isn't ideal as don't support simultaneous users/calls. But for now I think it's a step forward to connect the client and server.

Reply:
@cguedes Agree this isn't ideal, but it's probably fine for now.

Alternatively, you could implement this was middleware: https://fastapi.tiangolo.com/tutorial/middleware/